Write NamespaceStatus endpoints test +3 Endpoints

riaankleinhans opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- readCoreV1NamespaceStatus
- patchCoreV1NamespaceStatus
- replaceCoreV1NamespaceStatus

**Which issue(s) this PR fixes:**
Fixes #109762

**Testgrid Link:** [testgrid](https://testgrid.k8s.io/sig-api-machinery-gce-gke#gce-serial&width=20&graph-metrics=test-duration-minutes&include-filter-by-regex=should.apply.changes.to.a.namespace.status)


**Special notes for your reviewer:**
Adds +2 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig testing
/sig architecture
/area conformance

Riaankl commented:
/triage accepted

Riaankl commented:
/hold
Wait for 1.25 release to open

Riaankl commented:
/test pull-kubernetes-e2e-kind-ipv6
Unreleated flakes

Riaankl commented:
/unhold
>code freeze for 1.24 has been lifted and both kubernetes/kubernetes and kubernetes/website are now thawed

Riaankl commented:
/assign @deads2k @caesarxuchao
Can we please have a review of this e2e test, with the purpose of promoting it to Conformance after it passes the 2 week flake free requirement. Cleaning up the last Conformance technical debt.

Riaankl commented:
/hold 
replaceCoreV1NamespaceStatus to be added soon
/assign @heyste

Riaankl commented:
/unhold
@heyste thank you for adding `replaceCoreV1NamespaceStatus`
PR ready for review and merge

Riaankl commented:
/assign @smarterclayton

Please review this PR as will. TIA

smarterclayton commented on test/e2e/apimachinery/namespace.go:
```diff
@@ -282,4 +286,74 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
+		})
+		framework.ExpectNoError(err, "failed to marshal unstructured response")
+
+		payload := "{\"status\": {\"conditions\": [" + string(patchStatus) + "]}}"
```
Is there a reason to use this construct vs what's in job status (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/job.go#L488) or vice versa?

Wasn't clear to me whether this was part of the test (subtly)

smarterclayton replied:
What sort of issues?  Surprising to me on namespace, so makes me curious.

heyste replied:
I think it's mostly my mental model of the problem. Also, as `namespace.status` only has `conditions   <[]Object>` or `phase        <string>` my preference was to test changes with a set of `conditions`. With a fresh set of eyes on the problem I've got an local copy of the test using `f.Clientset` but still patching/checking for a new set of `conditions`.

heyste replied:
I changed the test around to follow the existing tests/style as much as I can.

heyste committed:
Create e2e test for NamespaceStatus endpoints

e2e test validates the following 3 endpoints
- patchCoreV1NamespaceStatus
- readCoreV1NamespaceStatus
- replaceCoreV1NamespaceStatus

smarterclayton commented:
/lgtm
/approve

Riaankl commented:
/test pull-kubernetes-e2e-kind

Unreleated flakes in CSI tests

Riaankl commented:
Unrelated CSI Volume tests that fail

/test pull-kubernetes-e2e-kind
